### PR TITLE
New actor-zk.

### PR DESCRIPTION
**Pull request by martun**

### CODE REVIEW

File: include/nil/crypto3/zk/commitments/batched_commitment.hpp [hunk 1]
```diff
@@ -173,18 +176,25 @@ namespace nil {
                     }
 
                     void eval_polys() {
-                        for(auto const &[k, poly] : _polys) {
```
Comment (Zerg1996) [inline include/nil/crypto3/zk/commitments/batched_commitment.hpp:176]: It's look good as it was.

File: include/nil/crypto3/zk/commitments/detail/polynomial/proof_of_work.hpp [hunk 2]
```diff
@@ -110,26 +113,50 @@ namespace nil {
...
-                        result = integral_type(transcript.template challenge<FieldType>().data);
-                        return proof_of_work;
+
+                        transcript(pow_seed + (std::size_t)pow_value_offset);
+                        transcript.template challenge<FieldType>();
+                        return pow_seed + (std::size_t)pow_value_offset;
                     }
 
-                    static inline bool verify(transcript_type &transcript, value_type proof_of_work) {
+                    static inline bool verify(transcript_type &transcript, value_type const& proof_of_work) {
```
Comment (Zerg1996) [inline include/nil/crypto3/zk/commitments/detail/polynomial/proof_of_work.hpp:159]: const &proof_of_work

File: include/nil/crypto3/zk/commitments/detail/polynomial/proof_of_work.hpp [hunk 3]
```diff
@@ -110,26 +113,50 @@ namespace nil {
...
+                                [&transcript, &pow_seed, &challenge_found, &pow_value_offset](std::size_t pow_start, std::size_t pow_finish) {
+                                    std::size_t i = pow_start;
+                                    while ( i < pow_finish ) {
+                                        if (challenge_found) {
+                                            break;
+                                        }
+                                        transcript_type tmp_transcript = transcript;
+                                        tmp_transcript(pow_seed + i);
+                                        integral_type pow_result = integral_type(tmp_transcript.template challenge<FieldType>().data);
+                                        if ( ((pow_result & mask) == 0) && !challenge_found ) {
```
[the remark this answers is not in the document]
Comment (martun) [reply include/nil/crypto3/zk/commitments/detail/polynomial/proof_of_work.hpp:139]: Yes, you are right.

File: include/nil/crypto3/zk/commitments/batched_commitment.hpp (hunk 1 above)
Comment (martun) [reply include/nil/crypto3/zk/commitments/batched_commitment.hpp:176]: For some reason, I can not capture k and poly by reference in the following lambda, if they are names used in unpacking.

**Pull request merged**